### PR TITLE
update only dirty attributes fixed (incl. Test)

### PR DESCRIPTION
**Pull request by OliverMahn-flinc**
I fixed the issue that dirty fields checking was not enabled on updating a resource.

### CODE REVIEW

File: Spine/DeserializeOperation.swift [hunk 1]
```diff
@@ -203,7 +203,9 @@ class DeserializeOperation: NSOperation {
 			if let extractedValue: AnyObject = self.extractAttribute(serializedData, key: key) {
 				let formattedValue: AnyObject = self.valueFormatters.unformat(extractedValue, forAttribute: field)
 				resource.setValue(formattedValue, forField: field.name)
-			}
+            } else {
```
Comment (wvteijlingen) [inline Spine/DeserializeOperation.swift]: What happens when an attribute is not included in the JSON? Is a `nil` value set on the Spine resource or is it left as is?

File: Spine/Resource.swift [hunk 2]
```diff
@@ -152,6 +152,16 @@ public class Resource: NSObject, NSCoding {
 		originalValues[field] = value ?? NSNull()
 	}
 	
+    func isDirty() -> Bool {
```
Comment (wvteijlingen) [inline Spine/Resource.swift]: Probably better to have this a computed property `isDirty: Bool`.

File: Spine/DeserializeOperation.swift (hunk 1 above)
Comment (OliverMahn-flinc) [inline Spine/DeserializeOperation.swift]: Got the point ... will update the pull request

**Comment by OliverMahn-flinc** [comment]
@wvteijlingen I implemented isDirty as computed property and fixed the nil deserialization handling. What do you think about this one?

**Comment by OliverMahn-flinc** [comment]
So, now I did write tests for checking updating relationships only if dirty and implemented the check.

**Comment by wvteijlingen** [comment]
Thanks a lot! I'll see if I can merge `dirty-checking` into `master` in the coming days. Will probably not be an easy merge, since so much changed in the meantime.

**Comment by urbany** [comment]
Hi @wvteijlingen any update on merging this with master? Thanks in advance!

**Pull request merged**